File: fix_api.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
  if (!fs.existsSync(dir)) return;
  const stat = fs.statSync(dir);
  if (stat.isDirectory()) {
    const entries = fs.readdirSync(dir);
    for (const entry of entries) {
      walk(path.join(dir, entry));
    }
  } else if (dir.endsWith('.js')) {
    try {
      const content = fs.readFileSync(dir, 'utf8');
      if (content.includes('http://backend:3001')) {
        const newContent = content.replace(/http:\/\/backend:3001/g, 'http://localhost:3001');
        fs.writeFileSync(dir, newContent, 'utf8');
        console.log('Fixed: ' + dir);
      }
    } catch (e) {
      console.error('Error: ' + dir + ' - ' + e.message);
    }
  }
}

walk('/app/.next');
console.log('Done!');

File: fix_api.py
import os
import sys

def replace_in_file(filepath):
    with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
        content = f.read()
    if 'http://backend:3001' in content:
        new_content = content.replace('http://backend:3001', 'http://localhost:3001')
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Fixed: {filepath}")

for root, dirs, files in os.walk('/app/.next'):
    for filename in files:
        if filename.endswith('.js'):
            filepath = os.path.join(root, filename)
            try:
                replace_in_file(filepath)
            except Exception as e:
                print(f"Error in {filepath}: {e}")

print("Done!")
